### PR TITLE
Service setup

### PR DESCRIPTION
**Pull request by samlsmith424**
## Features
1. ExercisesController, ExerciseService, ExerciseFacade, ExercisePoro, and their tests

## Concerns
1. Did not make a serializer yet and would like to check in with FE before doing so

## Pre-Submittal Checklist
- [x] All tests pass on local
- [x] Visually confirm view functionality
- [x] No prys or save_and_open_page commands remain
- [x] Unnecessary commented code removed

## As a GIF, how does this PR make you feel?
gif description

### CODE REVIEW

File: app/services/exercise_service.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class ExerciseService
```
Comment (gregoryjflaherty) [inline app/services/exercise_service.rb:1]: Great job here - @SullyBirashk , You and I need to remember to get these keys on our local

File: spec/services/exercise_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        expect(exercise).to have_key(:name)
+        expect(exercise[:name]).to be_a(String)
+
+        expect(exercise).to have_key(:target)
+        expect(exercise[:target]).to be_a(String)
+      end
+    end
+  end
+end
```
Comment (gregoryjflaherty) [inline spec/services/exercise_service_spec.rb:32]: Thorough testing - love it!

**Review by gregoryjflaherty — APPROVED** [review]

File: app/services/exercise_service.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    response = conn.get("exercises")
+    parse_json(response)
+  end
+
+  private
+
+  def self.conn
+    Faraday.new(url: "https://exercisedb.p.rapidapi.com/") do |require|
+      require.headers['X-RapidAPI-Host'] = 'exercisedb.p.rapidapi.com'
+      require.headers['X-RapidAPI-Key'] = ENV['exercise_key']
```
Comment (gregoryjflaherty) [inline app/services/exercise_service.rb:13]: Great job here - @SullyBirashk , You and I need to remember to get these keys on our local

File: app/controllers/api/v1/exercises_controller.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class Api::V1::ExercisesController < ApplicationController
+  def index
+    exercises = ExerciseFacade.find_all_exercises #(params[:all])
+    # render json: ExerciseSerializer.new(exercises)
```
Comment (SullyBirashk) [inline app/controllers/api/v1/exercises_controller.rb:4]: Nice job preparing for the serializer even though it isn't created yet!!

File: app/poros/exercise_poro.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+class ExercisePoro
+  attr_reader :body_part,
+              :equipment,
+              :gif_url,
+              :id,
+              :name,
+              :target_muscle
+
+  def initialize(exercise)
```
Comment (SullyBirashk) [inline app/poros/exercise_poro.rb:9]: Appreciate the detailed names

File: app/services/exercise_service.rb (hunk 3 above)
Comment (SullyBirashk) [reply app/services/exercise_service.rb:13]: Thanks for Pointing that out @gregoryjflaherty

**Pull request merged**